File: app/src/main/java/workshop/akbolatss/tagsnews/di/component/BoardComponent.java
package workshop.akbolatss.tagsnews.di.component;

import dagger.Component;
import workshop.akbolatss.tagsnews.di.module.BoardModule;
import workshop.akbolatss.tagsnews.di.module.DetailsModule;
import workshop.akbolatss.tagsnews.di.scope.ActivityScope;
import workshop.akbolatss.tagsnews.screen.board.BoardActivity;


@ActivityScope
@Component(dependencies = AppComponent.class, modules = {BoardModule.class, DetailsModule.class})
public interface BoardComponent {

    void inject(BoardActivity activity);
}

File: app/src/main/java/workshop/akbolatss/tagsnews/di/component/BoardComponent.kt
package workshop.akbolatss.tagsnews.di.component

import dagger.Component
import workshop.akbolatss.tagsnews.di.module.BoardModule
import workshop.akbolatss.tagsnews.di.scope.ActivityScope
import workshop.akbolatss.tagsnews.screen.board.BoardActivity

/**
 * Dagger2 Component with specific scope for BoardActivity
 * @see BoardModule
 * @see BoardActivity
 */
@ActivityScope
@Component(dependencies = [(AppComponent::class)], modules = [(BoardModule::class)])
interface BoardComponent {

    fun inject(activity: BoardActivity)
}
